instrumentation/spring/spring-webflux/spring-webflux-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/webflux/v5_0/server/RouterFunctionInstrumentation.java

```diff
@@ -61,6 +65,26 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class RouteAdvice {
 
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void methodEnter(
+        @Advice.Argument(0) ServerWebExchange exchange,
+        @Advice.Argument(1) Object handler,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+
+      if (handler == null) {
+        return;
+      }
+
+      Context parentContext = Context.current();
+      if (!instrumenter().shouldStart(parentContext, handler)) {
+        return;
+      }
+
+      context = instrumenter().start(parentContext, handler);
+      scope = context.makeCurrent();
```

Comment:
The scope does not seem to be closed anywhere

Reply:
no wonder other tests kept failing. thank you!